src/tribler/core/components/restapi/rest/utils.py

```diff
@@ -66,3 +73,42 @@ def fix_unicode_array(arr):
             new_arr.append(item)
 
     return new_arr
+
+
+def shorten(s: Optional[str], width: int = 50, placeholder='[...]', cut_at_the_end: bool = True):
```

Comment:
I'd expect `width` to be the total width *include* placeholder, not that placeholder is added to that width.

Reply:
**nitpick** The `'[...]'` default placeholder looks a bit heavyweight to me; maybe we can just use `'...'`

Reply:
It is just your personal preference. https://docs.python.org/3/library/textwrap.html

Reply:
> I'd expect width to be the total width including the placeholder, not that the placeholder is added to that width.

Yes, that version is more accurate. However, I would argue that we don't necessarily need it. Please keep in mind that this correct version introduces significant complexity compared to the original one.

Reply:
It would be something like this:
```python
def shorten(s: Optional[str], width: int = 50, placeholder='[...]', cut_at_the_end: bool = True):
    """ Shorten a string to a given width.
        Example: shorten('hello world', 5) -> 'hello[...]'
    """
    max_length = width - len(placeholder)
    if max_length < 0:
        max_length = width
        placeholder = ''

    if s and len(s) > max_length:
        return f'{s[:max_length]}{placeholder}' if cut_at_the_end else f'{placeholder}{s[-max_length:]}'
    return s
```

Reply:
Also, it is possible to throw an exception:
```
if len(placeholder) > width:
    raise ValueError(f"Placeholder {placeholder!r} is too big for the specified width {width}")
max_length = width - len(placeholder)
```

But I agree it is not very important in our case not to exceed the specified width, so we can skip it.